flappy-bird.js

```diff
@@ -197,17 +258,35 @@ export class Bird extends Scene {
         const t_after_click = this.click_time === 0 ? 0 : t - this.click_time;
 
 
-        this.elapsed_time_before_game_start = this.game_start? this.elapsed_time_before_game_start:t; //keep track of the time before user begin to play
+        this.elapsed_time_before_game_start = this.game_start ? this.elapsed_time_before_game_start : t; //keep track of the time before user begin to play
 
         this.update_y(t_after_click);
         this.update_angle(t_after_click);
         const model_transform = matrix_transform.times(Mat4.translation(0, this.y, 0))
-                                                .times(Mat4.rotation(this.angle,1,0,0));
+            .times(Mat4.rotation(this.angle, 1, 0, 0));
         this.draw_bird(context, program_state, model_transform);
-        
-        this.starting_distance = 10; //the distance between first pipe and the bird
-        const pipe_pos = this.game_start? this.starting_distance - (t-this.elapsed_time_before_game_start) * this.game_speed: this.starting_distance;
-        const starting_pipe_model_transform = matrix_transform.times(Mat4.translation(0, 10, pipe_pos));
-        this.draw_all_pipe(context,program_state, starting_pipe_model_transform);
+
+
+        // draw three sets of pipes alternatively to avoid gap at render position change
+        const pipe_pos1 = this.game_start ?
+            (this.starting_distance - (t - this.elapsed_time_before_game_start) * this.game_speed) % (this.pipe_distance * this.pipe_num)
```

Comment:
It's better to abstract this calculation into a function to increase readability